src/GraphQL/Types/Scalars/BooleanGraphType.cs

```diff
@@ -9,15 +9,33 @@ namespace GraphQL.Types
     public class BooleanGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value) => ((value as BooleanValue)?.Value).Boxed();
+        public override object ParseLiteral(IValue value) => value switch
+        {
+            BooleanValue b => b.Value.Boxed(),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override object ParseValue(object value) => ValueConverter.ConvertTo(value, typeof(bool));
+        public override bool CanParseLiteral(IValue value) => value is BooleanValue || value is NullValue;
 
         /// <inheritdoc/>
-        public override bool CanParseLiteral(IValue value) => value is BooleanValue;
+        public override object ParseValue(object value) => value switch
+        {
+            bool _ => value,
+            null => null,
+            _ => ThrowValueConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override IValue ToAST(object value) => new BooleanValue((bool)value);
+        public override bool CanParseValue(object value) => value is bool || value == null;
+
+        /// <inheritdoc/>
+        public override IValue ToAST(object value) => value switch
+        {
+            bool b => new BooleanValue(b),
+            null => new NullValue(),
+            _ => null
```

Comment:
? `ScalarGraphType` throws on unknown value.

Reply:
> Returning null indicates a failed conversion

Which is actually broken; I fixed that.

In any case, I added another protected method for this exception.